packages/connect/src/base/entry.ts

```diff
@@ -290,10 +308,14 @@ export class EntryNodes extends EventEmitter {
       this.usedRelays = this.availableEntryNodes
         // select only those entry nodes with smallest latencies
         .slice(0, MAX_RELAYS_PER_NODE)
-        .map(
-          (entry: EntryNodeData) =>
-            new Multiaddr(`/p2p/${entry.id.toB58String()}/p2p-circuit/p2p/${this.peerId.toB58String()}`)
-        )
+        .map((entry: EntryNodeData) => {
+          return {
+            relayDirectAddress: entry.multiaddrs[0],
+            ourCircuitAddress: new Multiaddr(
+              `/p2p/${entry.id.toB58String()}/p2p-circuit/p2p/${this.peerId.toB58String()}`
```

Comment:
I guess it might make sense to have a helper function to constuct these strings as well.